rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/HistoryTransformers.scala

```diff
@@ -0,0 +1,193 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import coop.rchain.rspace.{
+  util,
+  Blake2b256Hash,
+  DeleteContinuations,
+  DeleteData,
+  DeleteJoins,
+  HotStoreAction,
+  InsertContinuations,
+  InsertData,
+  InsertJoins,
+  Serialize
+}
+import java.nio.charset.StandardCharsets
+import coop.rchain.rspace.internal._
+import scodec.Codec
+import scodec.bits.{BitVector, ByteVector}
+import coop.rchain.rspace.nextgenrspace.history.HistoryRepositoryImpl._
+
+object HistoryTransformers {
+
+  // TODO more readable if case class
+  type Result = (Blake2b256Hash, Option[PersistedData], HistoryAction)
+
+  final case class MeasureJournal(
```

Comment:
could we spend some more time on this design?
key, type, action type, value seems to be less wasteful in this context.